tests/functional/tests/engine/test_pp.py

```diff
@@ -62,49 +64,35 @@ def test_change_to_nhit_and_back_io_in_flight(pyocf_ctx):
     cache.add_core(core)
 
     # Step 2
-    completions = []
-    for i in range(2000):
-        comp = OcfCompletion([("error", c_int)])
-        write_data = Data(4096)
-        io = core.new_io(
-            cache.get_default_queue(), i * 4096, write_data.size, IoDir.WRITE, 0, 0
-        )
-        completions += [comp]
-        io.set_data(write_data)
-        io.callback = comp.callback
-        io.submit()
+    r = (
+        Rio()
+        .target(core)
+        .njobs(10)
+        .bs(Size.from_KiB(4))
+        .readwrite(ReadWrite.RANDWRITE)
+        .size(core_device.size)
+        .time_based()
+        .time(timedelta(minutes=1))
+        .qd(10)
+        .run_async()
+    )
 
     # Step 3
     cache.set_promotion_policy(PromotionPolicy.NHIT)
 
     # Step 4
-    for c in completions:
-        c.wait()
-        assert not c.results["error"], "No IO's should fail when turning NHIT policy on"
+    r.abort()
```

Comment:
I think you mean "wait()" rather than "abort()" here - maybe add wait() to the Rio class?